core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -157,8 +159,10 @@ private void initializeCatalogTables() throws InterruptedException, SQLException
             return true;
           }
 
-          LOG.debug("Creating table {} to store iceberg catalog", JdbcUtil.CATALOG_TABLE_NAME);
-          return conn.prepareStatement(JdbcUtil.CREATE_CATALOG_TABLE).execute();
+          LOG.debug(
+              "Creating table {} to store iceberg catalog tables", JdbcUtil.CATALOG_TABLE_NAME);
+          return conn.prepareStatement(JdbcUtil.createCatalogTableOrViewSqlStatement(true))
```

Comment:
I think the public interfaces can still be `JdbcUtil.createCatalogTableSql` and `JdbcUtil.createCatalogViewSql`.  Internally it can still call the common SQL by filling required variables.

Boolean passing looks little confusing as we need to read the signature to understand that true is for table or view.

Reply:
OK, I will do that (not sure these methods are actually used by end user though).

Reply:
> not sure these methods are actually used by end user though

True. Since previous SQL like `JdbcUtil.CREATE_CATALOG_TABLE` was package private. Maybe these methods can also be package private.

Reply:
Exactly, it's what I'm doing now for clarify 😄